pkg/cloud/services/eks/identity_provider.go

```diff
@@ -0,0 +1,146 @@
+	s.scope.V(2).Info("got states", "desired", desired, "current", current)
+
+	identityProviderPlan := identityprovider.NewPlan(clusterName, current, desired, s.EKSClient, s.scope.Logger)
+
+	procedures, err := identityProviderPlan.Create(ctx)
+	if err != nil {
+		s.scope.Error(err, "failed creating eks identity provider plane")
+		return fmt.Errorf("creating eks identity provider plan: %w", err)
+	}
+	s.scope.V(2).Info("computed EKS identity provider plan", "numprocs", len(procedures))
+
+	// Perform required operations
+	for _, procedure := range procedures {
+		s.scope.Info("Executing identity provider procedure", "name", procedure.Name())
+		if err := procedure.Do(ctx); err != nil {
+			s.scope.Error(err, "failed executing identity provider procedure", "name", procedure.Name())
+			return fmt.Errorf("%s: %w", procedure.Name(), err)
+		}
+	}
+
+	latest, err := s.getAssociatedIdentityProvider(ctx, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "getting associated identity provider")
+	}
+
+	if latest != nil {
+		s.scope.ControlPlane.Status.IdentityProviderStatus = v1alpha4.IdentityProviderStatus{
+			ARN:    aws.StringValue(latest.IdentityProviderConfigArn),
+			Status: aws.StringValue(latest.Status),
+		}
+
+		err := s.scope.PatchObject()
+		if err != nil {
+			return errors.Wrap(err, "updating identity provider status")
+		}
+	}
+
+	return nil
+}
+
+func (s *Service) getAssociatedIdentityProvider(ctx context.Context, clusterName string) (*identityprovider.OidcIdentityProviderConfig, error) {
+	list, err := s.EKSClient.ListIdentityProviderConfigsWithContext(ctx, &eks.ListIdentityProviderConfigsInput{
+		ClusterName: aws.String(clusterName),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// these is only one identity provider
+
+	if len(list.IdentityProviderConfigs) == 0 {
+		return nil, nil
+	}
+
+	providerconfig, err := s.EKSClient.DescribeIdentityProviderConfigWithContext(ctx, &eks.DescribeIdentityProviderConfigInput{
+		ClusterName:            aws.String(clusterName),
+		IdentityProviderConfig: list.IdentityProviderConfigs[0],
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	config := providerconfig.IdentityProviderConfig.Oidc
+
+	return &identityprovider.OidcIdentityProviderConfig{
+		ClientID:                   config.ClientId,
+		GroupsClaim:                config.GroupsClaim,
+		GroupsPrefix:               config.GroupsPrefix,
+		IdentityProviderConfigArn:  config.IdentityProviderConfigArn,
+		IdentityProviderConfigName: config.IdentityProviderConfigName,
+		IssuerURL:                  config.IssuerUrl,
+		RequiredClaims:             config.RequiredClaims,
+		Status:                     config.Status,
+		Tags:                       converters.MapPtrToMap(config.Tags),
+		UsernameClaim:              config.UsernameClaim,
+		UsernamePrefix:             config.UsernamePrefix,
+	}, nil
+}
+
+func (s *Service) convertSDKToIdentityProvider(in *v1alpha4.OIDCIdentityProviderConfig) *identityprovider.OidcIdentityProviderConfig {
```

Comment:
What do you think to moving this to the [converters package](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/converters/eks.go)?

Reply:
Ack